README.org

```diff
@@ -1407,6 +1424,74 @@ To use ~worf-define-key~ to make the definitions, =:definer 'worf= can be specif
 To use ~lpy-define-key~ to make the definitions, =:definer 'lpy= can be specified.
 
 * FAQ
+** Will general.el slow my initialization time?
+There is an open issue where I talk about improving performance when rewriting general.el (and a "no overhead" experiment), but using general.el should not cause a major increase in startup time if you follow the guidelines listed here (see #180 and #502 for the original issues).
+
+1. Don't use general.el's deferred keybinding functionality
+
+General.el borrows a generic keybinding deferring mechanism from ~evil-define-key~. If a specified keymap does not exist when ~general-define-key~ is called, it will repeatedly check if that keymap now exists after loading something new until it does (e.g. see issue #180). It is much more efficient to use ~with-eval-after-load~ or the =use-package= =:config= or =:general-config= keywords.
+
+#+begin_src emacs-lisp
+;; If you don't load evil (or any package) immediately, do one of these
+(with-eval-after-load 'evil
+  (general-def 'normal ";" #'evil-ex))
+
+(use-package evil
+  :general-config
+  ('normal ";" #'evil-ex))
+
+(use-package evil
+  :config
+  (general-def 'normal ";" #'evil-ex))
+
+;; not this!
+(general-def 'normal ";" #'evil-ex)
+#+end_src
```

Comment:
Maybe what I'm about to say doesn't actually fit in here, but I think a much more common situation where one may accidentally run into this is when simply using `:general` to map to a keymap belonging to the package. Maybe you want to give one of those as a "not this!" example as well 🤷🏼‍♂️

Reply:
Okay, I see you mention this below in point 3!